From ff24098f5f2c196bd119d31c31d49cc585399fce Mon Sep 17 00:00:00 2001
From: Mihails Strasuns <mihails.strasuns.contractor@sociomantic.com>
Date: Mon, 20 Jun 2016 14:44:46 +0300
Subject: [PATCH 08/17] Add less ambious stomping flag name

Old name is still supported for backwards compatibility
with already ported projects.
---
 posix.mak         |  2 +-
 src/rt/lifetime.d | 17 ++++++++++++++---
 2 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/posix.mak b/posix.mak
index 700ab7a5..12dca193 100644
--- a/posix.mak
+++ b/posix.mak
@@ -5,7 +5,7 @@
 
 # Don't assert on stomping prevention during tests or any other internal targets
 # See https://github.com/sociomantic/druntime/pull/14 for details
-export ALLOW_STOMPING_PREVENTION=1
+export ASSERT_ON_STOMPING_PREVENTION=0
 
 QUIET:=@
 
diff --git a/src/rt/lifetime.d b/src/rt/lifetime.d
index afacd651..8d1350f0 100644
--- a/src/rt/lifetime.d
+++ b/src/rt/lifetime.d
@@ -240,9 +240,20 @@ void stomping_prevention_trigger_nonpure ( ) nothrow
 {
     import core.stdc.stdlib : getenv;
     import core.stdc.string : strcmp;
-    char* flag = getenv("ALLOW_STOMPING_PREVENTION".ptr);
-    if (flag && (strcmp(flag, "0") == 0))
-        assert(false, "Stomping prevention has been triggerred");
+
+    char* flag = getenv("ASSERT_ON_STOMPING_PREVENTION".ptr);
+
+    if (flag)
+    {
+        if (strcmp(flag, "1") == 0)
+            assert(false, "Stomping prevention has been triggerred");
+    }
+    else
+    {
+        flag = getenv("ALLOW_STOMPING_PREVENTION".ptr);
+        if (strcmp(flag, "0") == 0)
+            assert(false, "Stomping prevention has been triggerred");
+    }
 }
 
 /**
-- 
2.11.0

